Engine/BaseAssets/Components/Postprocessing/Bloom/PixelShaderBloom.cs

```diff
@@ -0,0 +1,205 @@
+using SharpDX.Direct3D11;
+using System;
+using System.Windows.Documents;
+using System.Collections.Generic;
+
+namespace Engine.BaseAssets.Components.Postprocessing.Bloom
+{
+    public class PixelShaderBloom : PostProcessEffect
+    {
+        private const int MaxIterationsCount = 16;
+        private SharpDX.Direct3D11.Device device => GraphicsCore.CurrentDevice;
+
+        private Texture[] samplingsTextures;
+        private Sampler sampler;
+        private RasterizerState backCullingRasterizerState;
+        private BlendState additiveBlendState;
+
+        private ShaderPipeline downsampleBoxPipeline;
+        private ShaderPipeline upsampleBoxPipeline;
+        private ShaderPipeline prefilterPipeline;
+
+        private int samplingsCount;
+
+        public PixelShaderBloom(int screenWidth, int screenHeight, int iterationsCount = 5)
+        {
+            if(iterationsCount <= 0 && iterationsCount > MaxIterationsCount)
+                throw new ArgumentOutOfRangeException(nameof(iterationsCount));
+
+            samplingsCount = iterationsCount;
+
+            sampler = new Sampler(TextureAddressMode.Clamp, TextureAddressMode.Clamp, Filter.MinMagMipLinear);
+
+            backCullingRasterizerState = new RasterizerState(device, new RasterizerStateDescription()
+            {
+                FillMode = FillMode.Solid,
+                CullMode = CullMode.Back,
+                IsFrontCounterClockwise = true,
+                IsScissorEnabled = false,
+                IsAntialiasedLineEnabled = true,
+                IsDepthClipEnabled = true,
+                IsMultisampleEnabled = true
+            });
+
+            BlendStateDescription blendStateDesc = new BlendStateDescription()
+            {
+                AlphaToCoverageEnable = false,
+                IndependentBlendEnable = false
+            };
+
+            blendStateDesc.RenderTarget[0] = new RenderTargetBlendDescription(
+                true, 
+                BlendOption.One, 
+                BlendOption.One, 
+                BlendOperation.Add,
+                BlendOption.Zero, 
+                BlendOption.One, 
+                BlendOperation.Add, 
+                ColorWriteMaskFlags.All
+            );
+
+            additiveBlendState = new BlendState(device, blendStateDesc);
+
+            Shader screenQuadShader = AssetsManager.Shaders["screen_quad"];
+            downsampleBoxPipeline = new ShaderPipeline(new Shader[] { screenQuadShader, Shader.Create("BaseAssets\\Shaders\\Bloom\\bloom_downsample_box.fsh") });
+            upsampleBoxPipeline = new ShaderPipeline(new Shader[] { screenQuadShader, Shader.Create("BaseAssets\\Shaders\\Bloom\\bloom_upsample_box.fsh") });
+            prefilterPipeline = new ShaderPipeline(new Shader[] { screenQuadShader, Shader.Create("BaseAssets\\Shaders\\Bloom\\bloom_prefilter.fsh") });
+
+            Resize(screenWidth, screenHeight);
+        }
+
+        public override void Resize(int width, int height)
```

Comment:
Make resize protected, so it's more safe and user can't magically fuck up bloom effect by calling a public function

Reply:
Remove resize, because that is unnecessary now